src/main/java/org/terasology/launcher/repositories/JenkinsRepositoryAdapter.java

```diff
@@ -96,13 +97,17 @@
     private Optional<GameIdentifier> computeIdentifierFrom(Jenkins.Build jenkinsBuildInfo) {
         return Optional.ofNullable(client.getArtifactUrl(jenkinsBuildInfo, "versionInfo.properties"))
                 .map(client::requestProperties)
-                .map(versionInfo -> versionInfo.getProperty("displayVersion"))
-                .map(displayVersion -> {
+                .map(versionInfo -> {
+                    var displayVersion = versionInfo.getProperty("displayVersion");
+                    if (displayVersion == null) {
+                        return null;
+                    }
                     // The Jenkins adapter has to append the build number to the display version to ensure that the
                     // version string is unique for two different builds. This is necessary because the display version
                     // generated by the CI build is the same of subsequent builds...
-                    String versionString = displayVersion + "+" + jenkinsBuildInfo.number;
-                    return new GameIdentifier(versionString, buildProfile, profile);
+                    String buildVersion = displayVersion + "+" + jenkinsBuildInfo.number;
+                    Semver engineVersion = new Semver(versionInfo.getProperty("engineVersion"), Semver.SemverType.IVY);
```

Comment:
This will throw an exception if `engineVersion` is not present in the properties file, or if it is present but no valid semver. 
The method contract is based on an `Optional` - either you get a valid identifier or an empty optional, but the call-side is not prepared for an exception. To bring it in line with the current code we should surround this with a try-catch to return `null` in case of error.

Reply:
Every time I hear this I think it makes `Optional` worse than `@Nullable`. We got a `Result<T, Exception>` type yet?

Actually, `client.requestProperties` looks like it's asynchronous, so `Mono<GameIdentifier>` would be appropriate here.

anyway. I've pushed a change to avoid the exception.

Reply:
I don't think we have a `Result` or `Either` in plain Java, but it's added by numerous functional-style libraries, I think...

Probably making use of ReactiveX `Mono` is not such a bad call :thinking: